### PR TITLE
Fix #12667: Change to using a setting to determine if in prod

### LINKED ISSUE #12667: Move to using setting to determine when site is running in production
In a couple of places we do a check like `is_prod = request.get_host() == "www.mozilla.org"`. This will no longer work in the new infra since the Host header will be the internal origin host name instead of the public one from the CDN. We do have an environment variable/setting, `APP_NAME` that is consistent across the old and new infra and that will tell us which deployment it is. The value will be `bedrock-prod` for prod, and `bedrock-stage` for stage, etc.. We should switch our logic to use this setting instead of the hostname. 

We discovered this issue because one of the places we do this check is for the `/robots.txt` URL where we disallow all crawling on all deployments except for production.

### PR DESCRIPTION
**Pull request by pmac**

### CODE REVIEW

File: bedrock/settings/base.py [hunk 1]
```diff
@@ -543,6 +543,7 @@ def get_app_name(hostname):
 CLUSTER_NAME = config("CLUSTER_NAME", default="")
 ENABLE_HOSTNAME_MIDDLEWARE = config("ENABLE_HOSTNAME_MIDDLEWARE", default=str(bool(APP_NAME)), parser=bool)
 ENABLE_VARY_NOCACHE_MIDDLEWARE = config("ENABLE_VARY_NOCACHE_MIDDLEWARE", default="true", parser=bool)
+IS_PROD = APP_NAME == "bedrock-prod"
```
Comment (stevejalim) [inline bedrock/settings/base.py:546]:
```suggestion
IS_MOZILLA_ORG = APP_NAME == "bedrock-prod"
```

Comment (pmac) [reply bedrock/settings/base.py:548]: I confirmed with brett that `APP_NAME == "bedrock-prod"` will be `True` for both pocket and mozorg. So I think maybe we can leave it? I added a comment.

**Comment by pmac** [comment]
Tests are now passing \o/

File: bedrock/settings/base.py (hunk 1 above)
Comment (stevejalim) [reply bedrock/settings/base.py:548]: OK, sure thing.

**Pull request merged**